### PR TITLE
Restricting access to S3 bucket by custom tag on the IAM identity according to security email in `databricks_aws_bucket_policy` data resource

### PR DESCRIPTION
**Pull request by edmondo1984**
This pull request add a restriction on the root bucket s3 policy, using the PrincipalTag of the request. An equivalent change should be implemented by customers who use SS3-KMS on their own IAM policies that are not managed by the Terraform plugin

Fixes https://github.com/databricks/terraform-provider-databricks/issues/1693

### CODE REVIEW

File: aws/data_aws_bucket_policy.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ func DataAwsBucketPolicy() *schema.Resource {
 						Principal: map[string]string{
 							"AWS": fmt.Sprintf("arn:aws:iam::%s:root", d.Get("databricks_account_id").(string)),
 						},
+						Condition: map[string]map[string]string{
```
Comment (nfx) [inline aws/data_aws_bucket_policy.go:39]: @edmondo1984 please make this field optional.

File: aws/data_aws_bucket_policy.go [hunk 2]
```diff
@@ -36,6 +36,11 @@ func DataAwsBucketPolicy() *schema.Resource {
...
+							"StringEquals": {
+								"aws:PrincipalTag/DatabricksAccountId": d.Get("databricks_account_id").(string),
```
Comment (nfx) [inline aws/data_aws_bucket_policy.go:41]: > Get your databricks-account-id. Go to the [account console](https://protect-us.mimecast.com/s/GK-NCOY0xwuNywrCER2IH?domain=docs.databricks.com) and click the in the bottom left corner. Under Account ID, select and copy the ID.

documentation meant the `E2 account id`, but in this data resource the `databricks_account_id` is AWS Account ID of Databricks

**Review by nfx — CHANGES_REQUESTED** [review]

**Comment by edmondo1984** [comment]
> @edmondo1984 could you also add documentation? :)

Sure, is the only thing that's missing?

**Comment by nfx** [comment]
@edmondo1984 yes, otherwise how users will discover this change? :)

**Review by nfx — APPROVED** [review]

**Pull request merged**